src/db_adapters/lists.py

```diff
@@ -0,0 +1,80 @@
+from typing import Literal, overload
+
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.db_adapters.media import ensure_media
+from src.db_tables.user_list import UserList, UserListItem, UserListItemKind
+
+
+@overload
+async def list_put_item(
+    session: AsyncSession,
+    user_list: UserList,
+    tvdb_id: int,
+    kind: Literal[UserListItemKind.MOVIE, UserListItemKind.SERIES],
+) -> UserListItem: ...
+
+
+@overload
+async def list_put_item(
+    session: AsyncSession,
+    user_list: UserList,
+    tvdb_id: int,
+    kind: Literal[UserListItemKind.EPISODE],
+    series_id: int,
+) -> UserListItem: ...
+
+
+async def list_put_item(
+    session: AsyncSession, user_list: UserList, tvdb_id: int, kind: UserListItemKind, series_id: int | None = None
+) -> UserListItem:
+    """Add an item to a user list, raises an error if the item is already present."""
```

Comment:
Document what error & probably also use the custom Value error like in the previous case